bootstrapsavagexr.sh

```diff
@@ -0,0 +1,290 @@
+
+	# Check that wget did not error..
+	#
+	if [ $? -ne 0 ]; then
+		banner "An ERROR occurred whilst fetching the installer." "Removing remains of installer file (if it was downloaded at all)...."
+		rm "${INSTALLER}"
+
+		exitScript
+	fi
+
+	# SHA256 check for download corruption (more common than you think...)
+	#
+	echo "${INSTALLER_SHA256}  ${INSTALLER}" | sha256sum --check
+
+	if [ $? -ne 0 ]; then
+		banner "An ERROR occurred whilst fetching the installer." "SHA256 Checksum mismatch, can occur if download got corrupted" "Removing remains of installer file...."
+		rm "${INSTALLER}"
+
+		exitScript
+	fi
+
+	chmod +x "${INSTALLER}"
+}
+
+#
+# ----------------------------------------------------------------------
+# runInstaller() Prepare for the Savage XR installer (create appropriate
+# directories) and run it.
+#
+runInstaller(){
+	banner "Preparing for installer, creating symbolic links etc"
+
+	# Make symbolic link from where Savage XR defaults its install directory
+	# to where it can persisted in flatpak, which is a stupidly long path that the
+	# user will no doubt get wrong, hence the symbolic link.
+	#
+	mkdir "${INSTALL_LOCATION}/${INSTALLER_DIR_NAME}"
+	cd ~
+	ln -s -T "${INSTALL_LOCATION}/${INSTALLER_DIR_NAME}" "${INSTALLER_DIR_NAME}"
+
+	banner "Running Savage XR installer, please choose default location." "But DE-SELECT:" "	* Launch Savage XR (as this installer needs to run further)" "	* Create menu Shortcuts (not needed)" "	* Create Desktop Shortcut (not needed)" "Then click on the 'Finish' button."
+
+	"${INSTALLER}"
+}
+
+#
+# ----------------------------------------------------------------------
+# cleanLibs() Remove library files which stop Savage XR from running under Flatpak,
+# this is compounded by the Savage XR Updater which puts them back!.
+#
+cleanLibs(){
+
+	for sLibraryName in "${aDeleteLibraries[@]}"
+	do
+	
+echo "Checking: '${INSTALL_LOCATION}/${INSTALLER_DIR_NAME}/${sLibraryName}'"
+		if [ -f "${INSTALL_LOCATION}/${INSTALLER_DIR_NAME}/${sLibraryName}" ]; then
+echo "Removing: '${INSTALL_LOCATION}/${INSTALLER_DIR_NAME}/${sLibraryName}'"
+			rm "${INSTALL_LOCATION}/${INSTALLER_DIR_NAME}/${sLibraryName}"
+			
+		fi
+
+	done
+}
+
+#
+# ----------------------------------------------------------------------
+# restoreSavagePreferences() There is a hidden ~/.savage directory which
+# contains persistent data logs and preferences, which would be lost on exist.
+# Not uber needed, but just in case...
+#
+restoreSavagePreferences(){
+	banner "Restoring Savage preference/log directory."
+
+	# Make symbolic link from where Savage XR defaults its install directory
+	# to where it can persisted in flatpak, which is a stupidly long path that the
+	# user will no doubt get wrong, hence the symbolic link.
+	#
+	if [ ! -d "${INSTALL_LOCATION}/${PREFERENCE_DIR}" ]; then
+		mkdir "${INSTALL_LOCATION}/${PREFERENCE_DIR}"
+	fi
+
+	cd ~
+	if [ ! -d "${PREFERENCE_DIR}" ]; then
```

Comment:
I have some doubts if it works as expected.

Reply:
Care to expand?

So far its worked fine, Savage XR does not obey the XDG base directories, most of its user config files remain in its directory (like a windows application) but it also drops a .savage folder in the home directory where it dumps a log file. Savage XR seems to work fine without it.

Reply:
Then you need `--persist=.savage` in finish-args. That also means you haven't actually tested it yet since it was broken.

Reply:
Changed to using --persist,